Objects/typeobject.c

```diff
@@ -1031,25 +1031,29 @@ PyType_FromSpec_Alloc(PyTypeObject *type, Py_ssize_t nitems)
     /* note that we need to add one, for the sentinel and space for the
        provided tp-traverse: See bpo-40217 for more details */
 
-    if (PyType_IS_GC(type))
+    if (PyType_IS_GC(type)) {
         obj = _PyObject_GC_Malloc(size);
-    else
+    }
+    else {
         obj = (PyObject *)PyObject_MALLOC(size);
+    }
 
-    if (obj == NULL)
+    if (obj == NULL) {
         return PyErr_NoMemory();
-
-    obj = obj;
```

Comment:
Would you mind to keep this instruction? I love it :-D (I'm kidding you, please remove it!)

Reply:
Maybe we should add some `volatile` 😉